editor/src/components/navigator/left-pane/github-pane/index.tsx

```diff
@@ -610,14 +612,48 @@ const LocalChangesBlock = () => {
     setCommitMessage(null)
   }, [branch, originCommit])
 
+  const projectId = useEditorState(Substores.restOfEditor, (store) => store.editor.id, 'project id')
+
+  const assetChecksums = useEditorState(
+    Substores.github,
+    (store) => store.editor.githubChecksums,
+    'Github checksums',
+  )
+
+  const projectContents = useEditorState(
+    Substores.projectContents,
+    (store) => store.editor.projectContents,
+    'project contents',
+  )
+
   const triggerSaveToGithub = React.useCallback(() => {
-    if (repo != null && cleanedCommitBranchName != null && commitMessage != null) {
-      dispatch(
-        [EditorActions.saveToGithub(repo, cleanedCommitBranchName, commitMessage)],
-        'everyone',
+    if (
+      repo != null &&
+      cleanedCommitBranchName != null &&
+      commitMessage != null &&
+      projectId != null
+    ) {
+      void saveProjectToGithub(
+        projectId,
+        repo,
+        persistentModelForProjectContents(projectContents),
+        dispatch,
+        {
+          branchName: cleanedCommitBranchName,
+          commitMessage: commitMessage,
+          assetChecksums: assetChecksums ?? {},
+        },
       )
     }
-  }, [dispatch, repo, commitMessage, cleanedCommitBranchName])
+  }, [
```

Comment:
should we add a log here for debugging (which one of repo, cleanedCommitBranchName etc is null)

Reply:
sounds like a good idea

Reply:
https://github.com/concrete-utopia/utopia/pull/3213/commits/1eb3b76d74180a620fe871c77967b05c89c4a733